src/metrics/metrics.ts

```diff
@@ -350,3 +357,138 @@ export function getMateAndAssists(pgnMoveLine: string) {
     lastPieceMoved,
   };
 }
+
+export async function getGameWithMostMoves(games: FileReaderGame[]) {
+  let maxNumMoves = 0;
+  let gameWithMostMoves: FileReaderGame | null = null;
+  let gameLinkWithMostMoves = null;
+
+  for await (const game of games) {
+    const chess = new Chess();
+
+    chess.loadPgn(game.moves);
+    const numMoves = chess.history().length;
+
+    if (numMoves > maxNumMoves) {
+      maxNumMoves = numMoves;
+      gameWithMostMoves = game;
+      let site = game.metadata
+      .find((item) => item.startsWith('[Site "'))
+      ?.replace('[Site "', '')
+      .replace('"]', '');
+      gameLinkWithMostMoves = site;
+    }
+  }
+
+  return {
+    gameLinkWithMostMoves,
+    maxNumMoves,
+  };
+
+}
+
+export async function getPieceLevelMoveInfo(games: FileReaderGame[]) {
+  const numMovesByPiece = {};
+  let averageNumMovesByPiece = {};
+  let pieceWithMostMovesInAGame = null;
+  let pieceWithHighestAverageNumMoves = null;
+  let gameLinkWithPieceMostMoves = null;
+  let numMovesMadePieceWithMostMoves = 0;
```

Comment:
NB/opinion but these are massive var names. Maybe we can call it 
```suggestion
  let mostInAGamePiece = null;
  let highestAveragePiece = null;
  let gameWithMostMoves = null;
  let maxMoves = 0;
```

Reply:
also what if there is a tie?

Reply:
less likely for large datasets and average, but p likely for mostMoves

Reply:
good point, will work on this

Reply:
i prefer the variable names i chose tbh. the ones you chose in my opinion would be overly simplistic to the point of confusion for me

Reply:
imo 7 words is too long